dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -151,6 +151,49 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
     */
   def project[U](implicit projector: SmartProject[T,U]): TypedDataset[U] = projector.apply(this)
 
+  /** Returns a new [[TypedDataset]] that contains the elements of both this and the `other` [[TypedDataset]]
+    * combined.
+    *
+    * Note that, this function is not a typical set union operation, in that it does not eliminate
+    * duplicate items. As such, it is analogous to `UNION ALL` in SQL.
+    *
+    * Differs from `Dataset#union` by aligning fields if possible.
+    * It will not compile if `Datasets` have not compatible schema.
+    *
+    * Example:
+    * {{{
+    *   case class Foo(x: Int, y: Long)
+    *   case class Bar(y: Long, x: Int)
+    *   case class Faz(x: Int, y: Int, z: Int)
+    *
+    *   foo: TypedDataset[Foo] = ...
+    *   bar: TypedDataset[Bar] = ...
+    *   faz: TypedDataset[Faz] = ...
+    *
+    *   foo union bar: TypedDataset[Foo]
+    *   foo union faz: TypedDataset[Foo]
+    *   // won't compile, you need to reverse order, you can't project from more fields to less
+    *   faz union foo
+    *
+    * }}}
+    *
+    * apache/spark
+    */
+  def union[U: TypedEncoder](other: TypedDataset[U])(implicit projector: SmartProject[U, T]): TypedDataset[T] =
+    TypedDataset.create(dataset.union(other.project[T].dataset))
+
+  /** Returns a new [[TypedDataset]] that contains the elements of both this and the `other` [[TypedDataset]]
+    * combined.
+    *
+    * Note that, this function is not a typical set union operation, in that it does not eliminate
+    * duplicate items. As such, it is analogous to `UNION ALL` in SQL.
+    *
+    * apache/spark
+    */
+  def union(other: TypedDataset[T]): TypedDataset[T] = {
```

Comment:
@Avasil , this was in a different file, any particular reason it was moved here (I like it more here too, just wondering if there was any other reason we did it)?

Reply:
@imarios I've figured that this new union isn't just forwarding method call so it shouldn't be in ```TypedDatasetForwarded``` and it wasn't compiling correctly when there were two methods with the same name in different place (I assume compiler thought it's an override)

Reply:
Ok, that's what I though. union was also giving issue for me if it was in the trait.